metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -933,9 +1006,24 @@ def gen_view_dataset_workunits(
             aspect=view_properties_aspect,
         ).as_workunit()
 
+    def gen_snapshot_dataset_workunits(
+        self,
+        table: BigqueryTableSnapshot,
+        columns: List[BigqueryColumn],
+        project_id: str,
+        dataset_name: str,
+    ) -> Iterable[MetadataWorkUnit]:
+        yield from self.gen_dataset_workunits(
+            table=table,
+            columns=columns,
+            project_id=project_id,
+            dataset_name=dataset_name,
+            sub_types=[DatasetSubTypes.TABLE],
```

Comment:
I know someone else said otherwise, but I think this should have a "Snapshot Table" subtype instead of the standard "Table"